packages/x-data-grid/src/utils/styled/css.ts

```diff
@@ -0,0 +1,102 @@
+import * as React from 'react';
+import { useTheme } from '@mui/material';
+import { applyStyled } from '@mui/system/createStyled';
+import type { CSSObject } from '@mui/system';
+import { injectStyles } from './engine';
+
+type Meta = {
+  name: string;
+  slot: string;
+};
+
+type StyleMeta<Keys extends string | number | symbol> = {
+  meta: Meta,
+  styles: { [key in Keys]: CSSObject },
+  classes: { [key in Keys]: string },
+};
+
+export function css<T extends Record<string, CSSObject>>(meta: Meta, styles: T): StyleMeta<keyof T> {
```

Comment:
The meta should come last, no? I imagine it is optional, and the styles are always defined.

Reply:
I'd leave it as it is for now, we can still evolve it if needed. Maybe I'd consider adding a 2nd declaration like this instead:

```typescript
function css(meta: M, styles: S);
function css(styles: S);
```

Having `meta` at the start keeps the style declarations more readable.

Reply:
No hard preference, but it is awkward to have the first param optional, while the second is required.

Reply:
It is, but I think having meta at the end would hurt readability too much.  
At least it can't be worst that lua's [`table.insert(list, [pos,] value)`](https://www.lua.org/pil/19.2.html) optional param *in the middle* lol.


I have thought about introducting separate functions instead, e.g. `cssSlot` and `css`.

Reply:
Having different functions would be even better in my opinion. This is too Material UI related (as we have styled from emotion and styled form MUI :) )